charts/external-dns/templates/deployment.yaml

```diff
@@ -23,8 +23,11 @@ spec:
       {{- with .Values.podLabels }}
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.podAnnotations }}
       annotations:
```

Comment:
You will need to check if `annotations` is needed as you might not have a secret or pod annotations.

Reply:
Whats the downside of an empty `annotations:` inside the generated templated?